pkg/model/core/decodable.go

```diff
@@ -23,33 +23,42 @@ import (
 type Decodable[T any] struct {
 	Value T
 
-	// TODO: Considering that these parameters are at most
-	// aes.BlockSize bytes in size, would it make sense to replace
-	// this with a fixed size array?
-	decodingParameters string
+	// Lets use a fixed-size array for decoding parameters, so Decodable[] can be marshalled, and used as a key in a map.
+	// We know actual decoding parameters are always <=16 length.
+	decodingParameters       [16]byte
+	decodingParametersLength uint8
 }
 
 // NewDecodable is a helper function for creating instances of
 // Decodable[T].
-func NewDecodable[T any](value T, decodingParameters []byte) Decodable[T] {
-	return Decodable[T]{
-		Value:              value,
-		decodingParameters: string(decodingParameters),
+func NewDecodable[T any](value T, decodingParameters []byte) (Decodable[T], error) {
+	var d Decodable[T]
+	d.Value = value
+	if len(decodingParameters) > 16 {
+		return d, status.Errorf(
+			codes.InvalidArgument,
+			"DecodingParameters is %d bytes in size, which exceeds the permitted maximum of 16 bytes",
+			len(decodingParameters),
+		)
 	}
+	d.decodingParametersLength = uint8(len(decodingParameters))
```

Comment:
Maybe a bit cleaner to write this as:
```go
d.decodingParametersLength = uint8(copy(...))
```

The `copy()` function returns the number of elements it copied.